### PR TITLE
Added Acceptance Tests

### PR DESCRIPTION
**Pull request by clintonb**
Tests for:
- Login via OpenID Connect
- Logistration+Enrollment

@stephensanchez @rlucioni

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,9 +12,10 @@ help:
 	@echo '    make html_coverage                        generate and view HTML coverage report         '
 	@echo '    make quality                              run pep8 and pylint                            '
 	@echo '    make validate                             run unit tests, followed by quality checks     '
+	@echo '    make accept                               run acceptance tests                           '
 	@echo '                                                                                             '
 
-dev_requirements:
+dev_requirements: test_requirements
 	pip install -qr requirements/local.txt --exists-action w
```
Comment (rlucioni) [inline Makefile:19]: Let's replace the `dev_requirements` and `test_requirements` targets with a single `requirements` target which installs requirements for testing and local dev.

File: acceptance_tests/api.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import requests
+from requests.auth import AuthBase
+
+from acceptance_tests.config import ENROLLMENT_API_URL, ENROLLMENT_API_TOKEN, ECOMMERCE_API_SERVER_URL, \
+    ECOMMERCE_API_TOKEN
```
Comment (rlucioni) [inline acceptance_tests/api.py]: [PEP 328](https://www.python.org/dev/peps/pep-0328/#rationale-for-parentheses) recommends the use of parentheses for long import statements.

File: acceptance_tests/api.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+class BearerAuth(AuthBase):
+    """ Attaches Bearer Authentication to the given Request object. """
+
+    def __init__(self, token):
+        """ Instantiate the auth class. """
+        self.token = token
+
+    def __call__(self, r):
+        """ Update the request headers. """
+        r.headers['Authorization'] = 'Bearer {0}'.format(self.token)
```
Comment (rlucioni) [inline acceptance_tests/api.py]: I don't think you need the `0` within the curly braces here.

File: acceptance_tests/api.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+class EnrollmentApiClient(object):
+    def __init__(self, host=None, key=None):
+        self.host = host or ENROLLMENT_API_URL
+        self.key = key or ENROLLMENT_API_TOKEN
+
+    def get_enrollment_status(self, username, course_id):
+        """
+        Retrieve the enrollment status for given user in a given course.
+        """
+        url = '{}/enrollment/{},{}'.format(self.host, username, course_id)
```
Comment (rlucioni) [inline acceptance_tests/api.py]: The use of placeholder strings within the curly braces might make this a little more readable.

File: README.rst [hunk 5]
```diff
@@ -32,6 +32,17 @@ Alternatively, you can launch the server using::
...
     $ ENABLE_DJANGO_TOOLBAR=1 ./manage.py runserver
 
+
+Testing
+-------
+Unit tests can be run with the following command:
+
+    $ make validate
+
+Acceptance tests require a valid LMS configuration and a user with a known username, email address, password, and access token. The following command will run the acceptance tests:
```
Comment (rlucioni) [inline README.rst:42]: Could you elaborate briefly and explain how one should go about retrieving an access token to use?

File: acceptance_tests/test_logistration_enrollment.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        orders = self.ecommerce_api_client.orders()
+        self.assertGreater(len(orders), 0, 'No orders found for the user!')
+
+        orders = sorted(orders, key=lambda o: o['date_placed'], reverse=True)
+        order = orders[0]
+
+        # TODO Find a better way to verify this is the correct enrollment.
+        # Verify the date and status
+        self.assertEqual(order['status'], 'Complete')
+        order_date = datetime.datetime.strptime(order['date_placed'], "%Y-%m-%dT%H:%M:%S.%fZ")
```
Comment (rlucioni) [inline acceptance_tests/test_logistration_enrollment.py]: Could you use `datetime.isoformat()` for this?

File: acceptance_tests/test_logistration_enrollment.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+        now = datetime.datetime.utcnow()
+        self.assertLess(order_date, now)
+
+        # Note: Much of this time is due to waiting on LMS page loads, especially when LMS is run on devstack.
+        self.assertGreater(order_date, now - datetime.timedelta(seconds=15))
```
Comment (rlucioni) [inline acceptance_tests/test_logistration_enrollment.py]: I think it'd be a little cleaner to pull this 15 second value out into a constant.

File: acceptance_tests/test_logistration_enrollment.py (hunk 6 above)
Comment (clintonb) [inline acceptance_tests/test_logistration_enrollment.py]: Nope. We are parsing a string in this case.

File: acceptance_tests/api.py (hunk 3 above)
Comment (clintonb) [inline acceptance_tests/api.py]: Removed.

File: acceptance_tests/api.py (hunk 4 above)
Comment (clintonb) [inline acceptance_tests/api.py]: Updated.

File: acceptance_tests/test_logistration_enrollment.py (hunk 6 above)
Comment (rlucioni) [inline acceptance_tests/test_logistration_enrollment.py]: Right. Using `dateutil.parser` might be a little cleaner than using `strptime`, but this is fine.

File: README.rst (hunk 5 above)
Comment (clintonb) [inline README.rst:42]: Updated.

File: acceptance_tests/test_logistration_enrollment.py (hunk 6 above)
Comment (clintonb) [inline acceptance_tests/test_logistration_enrollment.py]: I agree; but, I'd rather not include an entire library to replace a single call. If we do expand our usage of date parsing, we should definitely use dateutil.

File: acceptance_tests/test_logistration_enrollment.py (hunk 7 above)
Comment (clintonb) [inline acceptance_tests/test_logistration_enrollment.py]: Moved to `config.py`.

File: acceptance_tests/api.py (hunk 2 above)
Comment (clintonb) [inline acceptance_tests/api.py]: Updated.

**Comment by rlucioni** [comment]
:+1:

File: Makefile (hunk 1 above)
Comment (rlucioni) [inline Makefile:19]: @jimabramson is taking care of this over in #47.

**Pull request merged**